bokeh/sphinxext/bokeh_gallery.py

```diff
@@ -120,7 +121,7 @@ def config_inited_handler(app, config):
             continue
 
         with open(detail_file_path, "w") as f:
-            source_path = abspath(join(app.srcdir, "..", "..", detail["path"]))
+            source_path = abspath(expanduser(join(app.srcdir, "..", "..", detail["path"])))
```

Comment:
This is an extension for generating docs, the path is controlled and will never contain a `~`